pkg/reconciler/reconcile.go

```diff
@@ -19,5 +38,17 @@ func Reconcile(log logr.Logger, client *db.Database, req forman.Request) forman.
 	}
 	log.Info("Provider", "clusterId", provider.ClusterID)
 
-	return forman.Result{}
+	r := &fleetReconciler{
+		ctx:          ctx,
+		log:          log,
+		providerName: provider.ClusterID,
```

Comment:
The provider record is already loaded (from the DB) so we should save it on the reconciler not just the name 
```suggestion
		provider: provider,
```

Reply:
done.